src/common/functions/searchProf.ts

```diff
@@ -0,0 +1,38 @@
+import type { Universities, Professors } from "@prisma/client";
+import { db } from "@/server/db";
+
+export type SearchProfResult = {
+  uniAbbrv: Universities["abbrv"];
+  profName: Professors["name"];
+  profSlug: Professors["slug"];
+};
+
+// this is a band-aid solution
+// TODO: replace with better search algorithm
+export function searchProf(
+  query: string,
+  limit: number = 5,

```

Comment:
Question: Is there a reason for this limit?

Reply:
this is intentionally made to limit size of return so a search wouldnt take too long

allowing for loading more data either via manual or automatic means like infinite scrolling can be worked on as a separate task, similar to #144